media-api/app/lib/ImageExtras.scala

```diff
@@ -40,29 +40,46 @@ object ImageExtras {
   import scala.concurrent.duration._
   import com.gu.mediaservice.lib.FeatureToggle
 
-  def validityMap(image: Image): Map[String, Boolean] = Map(
-    "paid_image"           -> Costing.isPay(image.usageRights),
-    "conditional_paid"     -> Costing.isConditional(image.usageRights),
-    "no_rights"            -> !hasRights(image.usageRights),
-    "missing_credit"       -> !hasCredit(image.metadata),
-    "missing_description"  -> !hasDescription(image.metadata),
-    "current_deny_lease"   -> hasCurrentDenyLease(image.leases),
-    "over_quota"           -> UsageQuota.isOverQuota(image.usageRights)
+  case class ValidityCheck(valid: Boolean, overrideable: Boolean)
+
+  def validityMap(image: Image): Map[String, ValidityCheck] = Map(
+    "paid_image"           -> ValidityCheck(Costing.isPay(image.usageRights), true),
+    "conditional_paid"     -> ValidityCheck(Costing.isConditional(image.usageRights), true),
+    "no_rights"            -> ValidityCheck(!hasRights(image.usageRights), true),
+    "missing_credit"       -> ValidityCheck(!hasCredit(image.metadata), false),
+    "missing_description"  -> ValidityCheck(!hasDescription(image.metadata), false),
+    "current_deny_lease"   -> ValidityCheck(hasCurrentDenyLease(image.leases), true),
+    "over_quota"           -> ValidityCheck(UsageQuota.isOverQuota(image.usageRights), true)
   )
 
   def validityOverrides(image: Image, withWritePermission: Boolean): Map[String, Boolean] = Map(
     "current_allow_lease" -> hasCurrentAllowLease(image.leases),
     "has_write_permission" -> withWritePermission
   )
 
-  def invalidReasons(validityMap: Map[String, Boolean]) = validityMap
-    .filter(_._2 == true)
+  def invalidReasons(
+    validityMap: Map[String, ValidityCheck],
+    isOverriden: Boolean = false
+  ) = validityMap
+    .filter { case (_,v) => v.valid }
     .map { case (id, _) => id -> validityDescription.get(id) }
     .map {
       case (id, Some(reason)) => id -> reason
       case (id, None) => id -> s"Validity error: ${id}"
     }.toMap
 
-  def isValid(validityMap: Map[String, Boolean], validityOverrides: Map[String, Boolean]): Boolean =
-    !optToBool(validityMap.find(_._2 == true)) || optToBool(validityOverrides.find(_._2 == true))
+  def naiveIsValid(validityMap: Map[String, ValidityCheck]) =
+    !optToBool(validityMap.find(_._2.valid == true))
+
+  def invalidityIsOverridden(
+    validityMap: Map[String, ValidityCheck],
+    validityOverrides: Map[String, Boolean]
+  ): Boolean = {
+      optToBool(validityOverrides.find(_._2 == true)) &&
+      !optToBool(validityMap.filter(_._2.valid == true)
+        .find(_._2.overrideable == false))
+  }
```

Comment:
My instinct is adding a test is adding _more_ complicated code to test something already complicated. A test might help catch changes in behaviour if someone comes along and breaks it by accident.

I'd prefer to make the code clearer (and probably more verbose) to help with both things.